### ISSUE TITLE
Native Checkout Implematation for Wordpress+WooCommerce 

### ISSUE THREAD

**Issue by MasterJuan** [VivaPayments/API#43]
Hello,

I would like assistant in order to implement native checkout getaway as an option to woocommerce checkout page!

Thanks!

**Comment by dpnevmatikos** [comment]
Hi there @MasterJuan ,
You can find all the necessary documentation on how to integrate Native Checkout with your website [here](https://github.com/VivaPayments/API/wiki/Native-Checkout).

Feel free to ask if you have any specific questions during the integration process!

**Comment by MasterJuan** [comment]
I am trying to convert a free plugin which using redirect method in order to be able to enable Viva Payment gateway via Woocommerce dashobard and giving the appropriate credentials via custom fields. Even if, I made to print custom form as an extra option, I am not able to make it work!

Here is the staging url: https://gypeattarti.staging.wpengine.com

**Comment by MasterJuan** [comment]
In fact, I try to marry this plugin https://wordpress.org/support/plugin/woo-payment-gateway-viva-wallet and convert it to run on native checkout! Unsuccessfully so far. I can not make it to contact with Viva's API. In fact, how can I test whether or not contacts with the API, except the wait to see the trial payment to the dashboard of demo account?

**Comment by atserentzoulias** [comment]
Hi,
same issue.
The documentation is poor regarding native method.
We are trying to implement such solution to client website with no results.
because of the fact that the there is no other plugin in the market using that feature, we are currently trying to use the specific library with woocommerce.
Please provide detailed info.

**Comment by MasterJuan** [comment]
Could you please Enable https protocol for demo acounts?

Right now my website uses https protocol, so when i am trying to test the whole process using https, i receive this error "wrong user credentials or API connection".

I am trying to implement VivaWallet simple button checkout. However, I want to test it firstly in demo environment. So, I am using data-vp-baseurl="http://demo.vivapayments.com". When i use this url i can see the button but at my console i receive this message: 
"Failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('http://demo.vivapayments.com') does not match the recipient window's origin ('https://www.e-attarti.gr')."
When I change data-vp-baseurl to 'https://demo.vivapayments.com', I don't see the button and instead i received the message above "wrong user credentials or API connection".

**Comment by dpnevmatikos** [comment]
Hi there @MasterJuan,
Could you please check if the example below works at your web server ?
Just copy and paste it in an HTML file and then change ```data-vp-publicKey``` value
to your own.

```
<html>
<head>
    <meta name="viewport" content="width=device-width" />
    <title>SimpleCheckout Test Page</title>
</head>
<body style="background-color: cornflowerblue ">
    <form id="myform" action="myaction" method="post">
	<button type="button"
                data-vp-publickey="mykey"
                data-vp-amount="3321"
                data-vp-description="A description"
                data-vp-customeremail="mymail@mymail.com"
                data-vp-customerfirstname="First"
                data-vp-customersurname="Last"
                data-vp-baseurl="https://demo.vivapayments.com"
                data-vp-lang="el">
            Pay with SimpleCheckout
        </button>
    </form>

    <script src="https://code.jquery.com/jquery-1.11.2.js"></script>
    <script src="https://demo.vivapayments.com/web/checkout/js"></script>
</body>
</html>
```

**Comment by MasterJuan** [comment]
Hello @dpnevmatikos,
Yeah it works, however I am not able to see any transaction to my demo account!

Can you guide/help me to implement C# code inside a php file?
https://github.com/VivaPayments/API/wiki/Pay-with-Viva-Wallet

Is this the reason why I don't see anything at the sales section on demo account?

**Comment by dpnevmatikos** [comment]
You guessed correctly. The real reason for not seeing any transaction is because none exists, yet :) 
As stated in the docs : 

> The code above does not create charges – it only creates a token. The token has to be used for making the actual charge on your server.

What you need to do now is : get the payment Token that the form submits and make an HTTP call to the Viva Payments Api to finalize the payment. This is described in detail at [step 2 in the docs](https://github.com/VivaPayments/API/wiki/Pay-with-Viva-Wallet#step-2-make-the-charge).

Unfortunately I do not possess any knowledge of PHP so I can't help with the code.

**Comment by MasterJuan** [comment]
Alright, I made it to connect VivaWallet with my viva account and to receive back to my woocommerce dashboard all the info of the order.

I bored the code of a free plugin which uses  curl_setopt to send the information.
However, I am not sure what do a few lines of code, i.e. curl_setopt($session, CURLOPT_HEADER, false);

Using demo card values, I am able to see this info at my viva account and the pending status into my Woocommerce dashboard.

![vivapayments](https://cloud.githubusercontent.com/assets/3944059/13114679/3cf6f586-d59d-11e5-83c9-0a207f5bbce0.PNG)

After a few minutes the above notice disappears from my panel. Is any way to extend this "timeout"?
